File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/anoncreds/ProverStoreClaimOfferTest.java
package org.hyperledger.indy.sdk.anoncreds;

import org.hyperledger.indy.sdk.InvalidStructureException;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.isA;

import java.util.concurrent.ExecutionException;

public class ProverStoreClaimOfferTest extends AnoncredsIntegrationTest {

	@Test
	public void testProverStoreClaimOfferWorks() throws Exception {

		initCommonWallet();

		Anoncreds.proverStoreClaimOffer(wallet, issuer1GvtClaimOffer).get();
	}

	@Test
	public void testProverStoreClaimOfferWorksForInvalidJson() throws Exception {

		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		String claimOffer = String.format("{\"issuer_did\":\"%s\"}", issuerDid);

		Anoncreds.proverStoreClaimOffer(wallet, claimOffer).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/AnonCredsTests/ProverStoreClaimOfferTest.cs
using Hyperledger.Indy.AnonCredsApi;
using Hyperledger.Indy.Test.Util;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.AnonCredsTests
{
    [TestClass]
    public class ProverStoreClaimOfferTest : AnonCredsIntegrationTestBase
    {
        private const string WALLET_NAME = "commonWallet";
        private const string WALLET_KEY = "commonWalletKey";
        private Wallet _wallet;
        private string _walletName = "storeClaimOfferWallet";
        
        [TestInitialize]
        public async Task CreateWallet()
        {
            await WalletUtils.CreateWallet(WALLET_NAME, WALLET_KEY);
            _wallet = await WalletUtils.OpenWallet(WALLET_NAME, WALLET_KEY);
        }

        [TestCleanup]
        public async Task DeleteWallet()
        {
            if(_wallet != null)
                await _wallet.CloseAsync();

            await WalletUtils.DeleteWallet(WALLET_NAME, WALLET_KEY);
        }

       
        [TestMethod]
        public async Task TestProverStoreClaimOfferWorks()
        {
            var claimOffer = "{\"issuer_did\":\"NcYxiDXkpYi6ov5FcYDi1e\",\"schema_seq_no\":1 }";

            // TODO await AnonCreds.ProverStoreCredentialOfferAsync(_wallet, claimOffer);
        }


        [TestMethod]
        public async Task TestProverStoreClaimOfferWorksForInvalidJson()
        {
            var claimOffer = "{\"issuer_did\":\"NcYxiDXkpYi6ov5FcYDi1e\"}";

            // TODO
            //var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
            //    AnonCreds.ProverStoreCredentialOfferAsync(_wallet, claimOffer)
            //);
        }

        [TestMethod]
        public async Task TestProverStoreClaimOfferWorksForInvalidIssuerDid()
        {
            var claimOffer = "{\"issuer_did\":\"invalid_base58_string\",\"schema_seq_no\":1}";

            // TODO
            //var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
            //    AnonCreds.ProverStoreCredentialOfferAsync(_wallet, claimOffer)
            //);
        }

    }
}
